pkg/proxy/proxy.go

```diff
@@ -0,0 +1,212 @@
+package proxy
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"net/http/httputil"
+	"strings"
+	"time"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/registration-service/pkg/application"
+	"github.com/codeready-toolchain/registration-service/pkg/auth"
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+	"github.com/codeready-toolchain/registration-service/pkg/context"
+	crterrors "github.com/codeready-toolchain/registration-service/pkg/errors"
+	"github.com/codeready-toolchain/registration-service/pkg/log"
+	"github.com/codeready-toolchain/registration-service/pkg/proxy/namespace"
+	"github.com/codeready-toolchain/toolchain-common/pkg/cluster"
+
+	"github.com/gin-gonic/gin"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	controllerlog "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	ProxyPort = "8081"
+)
+
+type Proxy struct {
+	namespaces  *UserNamespaces
+	tokenParser *auth.TokenParser
+}
+
+func NewProxy(app application.Application, config configuration.RegistrationServiceConfig) (*Proxy, error) {
+	cl, err := newClusterClient()
+	if err != nil {
+		return nil, err
+	}
+	return newProxyWithClusterClient(app, config, cl)
+}
+
+func newProxyWithClusterClient(app application.Application, config configuration.RegistrationServiceConfig, cln client.Client) (*Proxy, error) {
+	// Initiate toolchain cluster cache service
+	cacheLog := controllerlog.Log.WithName("registration-service")
```

Comment:
Unfortunatly this specific log is used by the kube client which is used by the cluster service from common (the service which pulls the member clusters and caches them). It's not used by the rest of the proxy. We use the same log for the reg-service and for the rest of the proxy. I thought it would be confusing if some small part of the proxy uses one logger and the rest of the proxy uses a different one.